libraries/libcore/inc/status.h

```diff
@@ -82,5 +82,8 @@ void status_register_callback(StatusCallback callback);
 #define status_clear() status_msg(STATUS_CODE_OK, "Clear")
 
 // Use to forward failures or continue on success.
-#define status_ok_or_return(code) \
-  if (code) return (code)
+#define status_ok_or_return(code)          \
+  ({                                       \
+    __typeof__(code) status_expr = code;   \
+    if (status_expr) return (status_expr); \
```

Comment:
You don't need the parentheses around `status_expr` in the return statement, but you do need parentheses around `code` in the assignment since it's a macro argument; due to precedence issues with the literal copy/pasting the preprocessor does it's good practice to always put parentheses whenever you use a macro argument.

Also, nice job with the `__typeof__`!